apps/outlook-rooms-addin/src/tests/filter-space.component.spec.ts

```diff
@@ -0,0 +1,199 @@
+
+import {
+    mockOrgService,
+    mockSpacesService,
+    mockFeatureFilterService,
+    mockMapService,
+    mockRoomConfirmService,
+    mockRouterStub,
+    mockSpace,
+    mockSpaceWithViews,
+    mockEventFlowOptions,
+    mockFeatures,
+    mockGenerateEventForm,
+    mockForm,
+} from './test-mocks';
+
+describe('FindSpaceComponent', () => {
+    const formModel = mockForm;
+    const fb = new FormBuilder();
+    const form = fb.group(formModel);
+
+    const OrgServiceStub = mockOrgService;
+    const SpacesServiceStub = mockSpacesService;
+    const FeatureFilterServiceStub = mockFeatureFilterService;
+    const MapServiceStub = mockMapService;
+    const RoomConfirmServiceStub = mockRoomConfirmService;
+    const RouterStub = mockRouterStub;
+    let spectator: Spectator<FilterSpaceComponent>;
+
+    const createComponent = createComponentFactory({
+        component: FilterSpaceComponent,
+        imports: [
+            ReactiveFormsModule,
+            FormsModule,
+            MatFormFieldModule,
+            BookModule,
+            CommonModule,
+            BrowserModule,
+            BrowserAnimationsModule,
+            MatBottomSheetModule,
+            ComponentsModule,
+        ],
+        providers: [
+            {
+                provide: MatBottomSheet,
+                useValue: {
+                    open: jest.fn((FilterSpaceComponent) => {}),
+                    afterDismissed: jest.fn(),
+                },
+            },
+            {
+                provide: MAT_BOTTOM_SHEET_DATA,
+                useValue: {},
+            },
+            {
+                provide: MatBottomSheetRef,
+                useValue: {
+                    dismiss: jest.fn(),
+                },
+            },
+            {
+                provide: FeaturesFilterService,
+                useValue: {
+                    features$: of(mockFeatures),
+                    applyFilter: jest.fn(),
+                    getSelectedFeatures: jest.fn(),
+                },
+            },
+            {
+                provide: EventFormService,
+                useValue: {
+                    setView: jest.fn(() => {}),
+                    newForm: jest.fn(() => {}),
+                    clearForm: jest.fn(),
+                    storeForm: jest.fn(() => {}),
+                    loadForm: jest.fn(),
+                    postForm: jest.fn(),
+                    setOptions: jest.fn(() => {}),
+                    view: '',
+                    available_spaces: of(
+                        mockSpace,
+                        mockSpace,
+                        mockSpace,
+                        mockSpaceWithViews
+                    ),
+                    options: of(mockEventFlowOptions),
+                },
+            },
+        ],
+        declarations: [],
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+        jest.clearAllMocks();
```

Comment:
Not sure why you are creating the component twice and clearing the mocks before hand.
If you need to clean up after a test use `afterEach` instead.